### PR TITLE
Setup Typescript Bull Redis Queue

### PR DESCRIPTION
**Pull request by ericyliu**
Links
-----
- Issue: https://upchieve.monday.com/boards/438837796/pulses/518195818
- Redis: https://redis.io/topics/quickstart
- Bull: https://github.com/OptimalBits/bull

Description
-----------
- adds a typescript bull worker queue
- redis db expected to be local
- adds a new `.eslintrc.js` config to handle typescript files, moves the current rules to `eslint-javascript.js`

Developer self-review checklist
-------------------------------
- [ ] Potentially confusing code has been explained with comments
- [ ] No warnings or errors have been introduced; all known error cases have been handled
- [ ] Any appropriate documentation (within the code, README.md, docs, etc) has been updated
- [ ] All edge cases have been addressed

### CODE REVIEW

**Comment by ericyliu** [comment]
need to make eslint play nice with prettier/typescript

File: worker/index.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import * as Queue from 'bull';
+import { log } from './logger';
+import { workerQueueName, redisConnectionString } from '../config';
```
Comment (jkeat) [inline worker/index.ts:3]: Could you add these to `config.example.js`?

Comment (ericyliu) [reply worker/index.ts:3]: oh good catch, will do!

**Comment by jkeat** [comment]
To run locally, do you need to run `npm run worker:dev` and then `ts-node-dev scripts/add-cron-jobs`?

What are you envisioning for the deployment process? We could run the setup scripts in `bin/deploy` (that's what the slack chatbot triggers)

**Comment by jkeat** [comment]
With the ESLint VS Code extension, I'm seeing the typescript rules being applied to all the server files:

Screen Shot 2020-04-15 at 6 10 38 PM

Is this something that we'll need to fix with VS Code settings individually (or fix + commit the .vscode folder)?

**Comment by ericyliu** [comment]
@jkeat made the updates we talked about, looking at the deploy script, it seems like i need to start the worker with pm2, and the deploy script will just call `pm2 restart worker` to restart it upon fetching the updates

**Pull request merged**